mysql-backup-s3/backup.sh

```diff
@@ -39,74 +39,156 @@ if [ "${S3_IAMROLE}" != "true" ]; then
   export AWS_DEFAULT_REGION=$S3_REGION
 fi
 
-MYSQL_HOST_OPTS="-h $MYSQL_HOST -P $MYSQL_PORT -u$MYSQL_USER -p$MYSQL_PASSWORD"
-DUMP_START_TIME=$(date +"%Y-%m-%dT%H%M%SZ")
+my_mysql() {
+    mysql -h "$MYSQL_HOST" -P "$MYSQL_PORT" -u"$MYSQL_USER" -p"$MYSQL_PASSWORD" "$@"
+}
+
+dump_chosen () {
+    # ash shell does not support arrays and we need to get the mysql dump
+    # options in from the docker file so the least nasty way to do this is, to
+    # have an unquoted variable as far as I can see.  Make sure it doesn't come
+    # in from an untrusted source, which, in the original application I assume
+    # it can't.
+    # 
+    # shellcheck disable=SC2086
+    echo running mysqldump --host "$MYSQL_HOST" --port "$MYSQL_PORT" --user "$MYSQL_USER" --password="$MYSQL_PASSWORD" $MYSQLDUMP_OPTIONS --databases "$@" >&2
+    # shellcheck disable=SC2086
+    mysqldump --host "$MYSQL_HOST" --port "$MYSQL_PORT" --user "$MYSQL_USER" --password="$MYSQL_PASSWORD" $MYSQLDUMP_OPTIONS --databases "$@"
+}
+
+dump_all () {
+    # ash shell does not support arrays and we need to get the mysql dump
+    # options in from the docker file so the least nasty way to do this is, to
+    # have an unquoted variable as far as I can see.  Make sure it doesn't come
+    # in from an untrusted source, which, in the original application I assume
+    # it can't.
+    # 
+    # shellcheck disable=SC2086
+    echo running mysqldump --host "$MYSQL_HOST" --port "$MYSQL_PORT" --user "$MYSQL_USER" --password="$MYSQL_PASSWORD" $MYSQLDUMP_OPTIONS --all-databases >&2
+    # shellcheck disable=SC2086
+    mysqldump --host "$MYSQL_HOST" --port "$MYSQL_PORT" --user "$MYSQL_USER" --password="$MYSQL_PASSWORD" $MYSQLDUMP_OPTIONS --all-databases
+
+}
+
+
+dump_to_file() {
+    ( set -o pipefail
+    THE_DB=$1
+    THE_DUMP_FILE=$2
+    if [ "--all-databases"  = "$THE_DB" ]
+    then
+	dump_command=dump_all
+    else
+	dump_command=dump_chosen
+    fi
+    
+    if [ "${ENCRYPT}" = "false" ]
+    then
+	"$dump_command" "$THE_DB" | gzip > "$THE_DUMP_FILE"
```

Comment:
shouldn't all data extracted from a database be always encrypted to be safe?

Reply:
Right now I've taken the code from another open source project which doesn't encrypt.  I want to see if they are willing to take it back so I'm putting out a version which will be compatible with what they are doing.  Maybe we should eliminate the unencrypted branch from the code in the near future or do that in an internal branch?  I'll definitely think about this later once I see if they react to my feature request.